Makefile

```diff
@@ -2,8 +2,8 @@ export K8S_VERSION ?= 1.21.x
 export KUBEBUILDER_ASSETS ?= ${HOME}/.kubebuilder/bin
 
 ## Inject the app version into project.Version
-LDFLAGS ?= "-ldflags=-X=github.com/aws/karpenter/pkg/utils/project.Version=$(shell git describe --tags --always)"
-GOFLAGS ?= "-tags=$(CLOUD_PROVIDER) $(LDFLAGS)"
+LDFLAGS ?= -ldflags='-buildid= -extldflags=-zrelro -extldflags=-znow -X=github.com/aws/karpenter/pkg/utils/project.Version=$(shell git describe --tags --always)'
+GOFLAGS ?= -tags=$(CLOUD_PROVIDER) -trimpath -buildmode=pie $(LDFLAGS)
```

Comment:
Can you make sure that the rationale for these flags is clearly documented? It's pretty far outside of the expertise of the team.